src/library.tests/WorkloadInformationProviderTests.cs

```diff
@@ -30,19 +30,29 @@ public WorkloadInformationProviderTests()
         }
 
         [Theory]
-        [InlineData(1, 100)]
-        [InlineData(5, 20)]
-        [InlineData(10, 3)]
-        [InlineData(1, 3)]
-        public async void GetReachableServicesAsync_HeadlessService(int numServices, int numAddresses)
+        [InlineData(1, 100, true)]
+        [InlineData(5, 20, false)]
+        [InlineData(10, 3, true)]
+        [InlineData(2, 3, false)]
+        public async void GetReachableServicesAsync_HeadlessService(int numServices, int numAddresses, bool isInWorkloadNamespace)
         {
-            ConfigureHeadlessService(numServices: numServices, namingFunction: (i) => $"myapp-{i}", numAddresses: numAddresses, addressHostNamingFunction: (i) => $"Host-{i}");
-            var result = await _workloadInformationProvider.GetReachableEndpointsAsync(namespaceName: "", localProcessConfig: null, includeSameNamespaceServices: true, cancellationToken: default(CancellationToken));
+            var expected = ConfigureHeadlessService(numServices: numServices, namingFunction: (i) => $"myapp-{i}", numAddresses: numAddresses, addressHostNamingFunction: (i) => $"Host-{i}", isInWorkloadNamespace);
```

Comment:
can we change the variables expected to expectedServicesList and instead of result, finalServicesList since many of the variables are defined ```var``` it is confusing when someone reads the code. Just a nice to have suggestion. Other than than looks good, I like the idea of covering all different use cases programatically in the test using InlineData :). 

or another suggestion instead of var result, you can change it to ```IEnumerable<EndpointInfo> result ``` , same for expected variable. 

Thanks.

Reply:
sure, done.